apply/apply.go

```diff
@@ -62,14 +72,44 @@ func applyRepo(fs afero.Fs, p *plan.Plan, repoTemplates *packr.Box) error {
 		return e
 	}
 	// Remove after migration.
-
 	if !p.SiccMode {
 		e := fs.Remove(".sicc-version")
 		if e != nil {
 			log.Debug("error removing .sicc-version. ignoring")
 		}
 	}
-	return nil
+	return applyCustomProviders(fs, p)
+}
+
+func applyCustomProviders(fs afero.Fs, p *plan.Plan) (err error) {
+	for providerName, customProvider := range p.CustomProviders {
+		log.Infof("Applying custom provider %s", providerName)
+		err = customProvider.Install(providerName, fs)
+		if err != nil {
+			return errors.Wrapf(err, "Error applying custom provider %s", providerName)
+		}
+	}
+	return
+}
+
+// Terraform won't cache providers unless the directory structure already exists
+func applyCustomPluginCache(fs afero.Fs, p *plan.Plan, repoBox *packr.Box) error {
+	path := path.Join("", providers.CustomPluginCacheDir)
```

Comment:
Why do we need to join here?

Reply:
was joining to `rootPath` and forgot to rm it when i got rid of it